core/src/main/java/org/fao/geonet/kernel/batchedit/CSVBatchEdit.java

```diff
@@ -0,0 +1,377 @@
+
+			indexAndTaxonomy.indexReader.document(tdocs.scoreDocs[0].doc, docVisitor);
+			org.apache.lucene.document.Document doc = docVisitor.getDocument();
+
+			String uuid = doc.get("_uuid");
+			Log.debug(Geonet.SEARCH_ENGINE, "getMetadataByLuceneSearch --> uuid: " + uuid);
+
+			if (uuid != null) {
+				Metadata md = mdRepo.findOneByUuid(uuid);
+				return md;
+			}
+
+		} catch (Exception e) {
+			throw new BatchEditException("failed to get this record using lucene search");
+		}
+
+		return null;
+	}
+	
+	/**
+	 * Retrieves CRS reference system
+	 * @param crsId
+	 * @return
+	 */
+	public Crs getById(String crsId) {
+        for (Object object : ReferencingFactoryFinder
+            .getCRSAuthorityFactories(null)) {
+            CRSAuthorityFactory factory = (CRSAuthorityFactory) object;
+
+            try {
+                Set<String> codes = factory
+                    .getAuthorityCodes(CoordinateReferenceSystem.class);
+                for (Object codeObj : codes) {
+                    String code = (String) codeObj;
+                    if (code.equals(crsId)) {
+                        String authorityTitle = (factory.getAuthority()
+                            .getTitle() == null ? "" : factory
+                            .getAuthority().getTitle().toString());
+                        String authorityEdition = (factory.getAuthority()
+                            .getEdition() == null ? "" : factory
+                            .getAuthority().getEdition().toString());
+
+                        String authorityCodeSpace = "";
+                        Collection<? extends Identifier> ids = factory
+                            .getAuthority().getIdentifiers();
+                        for (Identifier id : ids) {
+                            authorityCodeSpace = id.getCode();
+                        }
+
+                        String description;
+                        try {
+                            description = factory.getDescriptionText(code)
+                                .toString();
+                        } catch (Exception e1) {
+                            description = "-";
+                        }
+                        description += " (" + authorityCodeSpace + ":" + code
+                            + ")";
+
+                        return new Crs(code, authorityTitle,
+                            authorityEdition, authorityCodeSpace,
+                            description);
+                    }
+                }
+            } catch (FactoryException e) {
+            }
+        }
+        return null;
+    }
+	
+	public String toTitleCase(String clval){
+		String[] vals = clval.split(" ");
+		StringBuilder titleCase = new StringBuilder();
+		if(vals.length > 1){
+			titleCase.append(vals[0].toLowerCase());
+			for(int i = 1; i < vals.length; i++){
+				titleCase.append(WordUtils.capitalize(vals[i]));
+			}
+		}
+		return titleCase.toString();
+	}
+}
+
+class Crs {
```

Comment:
This appears to be a duplication of [Crs](https://github.com/geonetwork/core-geonetwork/blob/master/services/src/main/java/org/fao/geonet/api/registries/model/Crs.java) class provided by GeoNetwork.

I would expect some justification for the duplication, and even then I would expect this class to extend the GeoNetwork `Crs` class to benefit from any changes and improvements over time.